### ISSUE TITLE
Add: AddDefaultIdentity()

### ISSUE THREAD

**Issue by HaoK** [aspnet/Identity#1625]
Based on discussions with @davidfowl 

Consider something like:

```
   services.AddIdentity<User, Role, DbContext>()
```

For auth consider adding a new options class with 4 `IOptionsBuilder` properties that allow configuration of each cookie.

cc @ajcvickers @blowdart

**Comment by HaoK** [comment]
Met with @davidfowl @Eilon @DamianEdwards and we decided to add a new

`AddDefaultIdentity<TUser>()` method which will collapse today's identity methods down except for the stores, roles will no longer be enabled by default as well.

Startup in the templates will now look like:
```
   services.AddDefaultIdentity<IdentityUser>.AddEntityFrameworkStores<ApplicationDbContext>();
```

Its will be functionally the same as:
```
            services.AddIdentityCore<IdentityUser>(o => o.Stores.MaxLengthForKeys = 128)
                 .AddEntityFrameworkStores<ApplicationDbContext>()
                 .AddDefaultUI()
                 .AddDefaultTokenProviders();

            services.AddAuthentication(o =>
            {
                o.DefaultScheme = IdentityConstants.ApplicationScheme;
                o.DefaultSignInScheme = IdentityConstants.ExternalScheme;
            })
            .AddIdentityCookies(o => { });
```

The rest of auth changes will be looked at in 2.2

**Comment by raffaeler** [comment]
Could you please confirm that RTM implementation does not require the following anymore?
```
services.AddAuthentication(o =>
            {
                o.DefaultScheme = IdentityConstants.ApplicationScheme;
                o.DefaultSignInScheme = IdentityConstants.ExternalScheme;
            })
            .AddIdentityCookies(o => { });
```
I get an error of scheme being already registered...

**Comment by HaoK** [comment]
You need to add roles back, it isn't enabled by default anymore `.AddRoles<IdentityRole>`

**Comment by Rick-Anderson** [comment]
>You need to add roles back, it isn't enabled by default anymore `.AddRoles<IdentityRole>`

This was fixed in #1832 so roles are now added by by `AddDefaultIdentity`
See https://github.com/aspnet/Identity/issues/1813

**Comment by ntimmerman** [comment]
This is a joke.
